src/app/forms/components/checkbox-group/checkbox-group.component.ts

```diff
@@ -15,33 +15,40 @@ import { BaseControlComponent } from '../base-control/base-control.component';
   ]
 })
 export class CheckboxGroupComponent extends BaseControlComponent {
```

Comment:
How are you handling edit mode? i.e. a value is already loaded and correspondent checkboxes need to be selected. I can see that there is `isChecked` function that is been called from the template but there is no way to set the value from the outside or is there?

Reply:
the `writeValue` function propagates the form control value to the template through `ngModel`. In the case of this component, because the value is an array (or string representation of one), is checked links each item with its respective checkbox. Because of this, external changes to the form would also update the form. e.g. `control.patchValue(...)`